### DISCUSSION TITLE
SeaweedFS as a distributed cache store for small files

### DISCUSSION THREAD

**Discussion by jfcalvo** [seaweedfs/seaweedfs#2848] [Q&A]
Hi!,

I'm investigating a solution for a file caching service that we need for one of our products.

The idea is to start small with one unique server with one or two hard disks and grow with more disks and more servers in the future if the application needs it.

The details of the solution that we want to implement:
* It's only a cache of files downloaded from different sources so not sure if replication is a good idea. If a cache file is lost it can be downloaded again.
* The size of files is in the range of 100KB to 60MB as much (SeaweedFS performing well with small files caught our attention).
* We need to connect the system to 4 different servers (that needs access to the cached files) but this could grow to 10, 20, or even more in the future.
* The system needs to be mounted on the different servers as a POSIX file system.
* Distributed locking for files is desirable, as mechanism to avoid multiple servers/processes downloading the same files at the same time (if one server starts downloading one file the other should wait until finished using locks).
* Files should be regularly deleted if they have not been accessed in a given amount of weeks.

We have been taking a look to SeaweedFS documentation but we are not sure if we are trying to overcomplicate things given that we only need to use the service to store cache files.

So my question; is SeaweedFS a good solution for a service like the one described?.

Thanks.

**Comment by kmlebedev** [comment]
Can fit with minimal modifications

**Reply by jfcalvo** [reply]
Please can you elaborate?. What minimal modifications?.

**Comment by chrislusf** [comment]
> * Distributed locking for files is desirable, as mechanism to avoid multiple servers/processes downloading the same files at the same time (if one server starts downloading one file the other should wait until finished using locks).

Distributed locking is not included. You may need to use a separate system for locking.

> * Files should be regularly deleted if they have not been accessed in a given amount of weeks.

Files are TTLed based on creation time, not by access time.

**Reply by jfcalvo** [reply]
> Distributed locking is not included. You may need to use a separate system for locking.

It is somehow in the roadmap as feature?

> Files are TTLed based on creation time, not by access time.

I guess I can use some utilities for this task of deleting by access time and add them to cron.

**Reply by chrislusf** [reply]
> It is somehow in the roadmap as feature?

Not planned.

**Reply by jfcalvo** [reply]
Ok, thank you very much for your answers @chrislusf

**Comment by jfcalvo** [comment]
@chrislusf I have another question. Does the access time works fine with the file POSIX adapter? I'm thinking into running a utility that will check if the access time is older than some specific time and in that case delete the file.

**Reply by chrislusf** [reply]
Access time is not tracked through.

**Reply by jfcalvo** [reply]
It would be difficult to implement? Thinking into take a look to the code and maybe create a PR.
